File: src/jk_5/nailed/map/gameloop/instructions/InstructionMoveTeamspeak.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;
import jk_5.nailed.players.Player;
import jk_5.nailed.teams.Team;
import jk_5.nailed.teamspeak3.TeamspeakClient;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class InstructionMoveTeamspeak implements IInstruction {

    private String team;
    private int channelID;

    @Override
    public void injectArguments(String arguments) {
        String data[] = arguments.split(" ", 2);
        this.team = data[0];
        this.channelID = Integer.parseInt(data[1]);
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Team team = Nailed.teamRegistry.getTeam(this.team);
        for (Player p : Nailed.playerRegistry.getPlayers()) {
            if(p.getTeam() != team) return;
            TeamspeakClient ts = p.getTeamspeakClient();
            if(ts == null) continue;
            Nailed.teamspeak.moveClientToChannel(ts, this.channelID);
        }
    }
}

File: src/jk_5/nailed/map/gameloop/instructions/InstructionMoveTeamspeak.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction
import jk_5.nailed.teamspeak3.TeamspeakManager

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionMoveTeamspeak extends IInstruction {
  private var team: String = null
  private var channelID: Int = 0

  def injectArguments(arguments: String) {
    val data = arguments.split(" ", 2)
    this.team = data(0)
    this.channelID = Integer.parseInt(data(1))
  }

  def execute(controller: GameThread) {
    val team = controller.getMap.getTeamManager.getTeam(this.team)
    if(team.isEmpty) return
    controller.getMap.getPlayers.filter(p => p.getTeam.eq(team.get) && p.getTeamspeakName != null).map(p => TeamspeakManager.getClientForUser(p.getTeamspeakName).get).foreach(ts => TeamspeakManager.moveClientToChannel(ts, this.channelID))
  }
}
